tests/modes.spec.ts

```diff
@@ -1,25 +1,132 @@
+  await clickMap(page, 500, 500);
+
+  await page.getByRole("button", { name: "1) Untitled point" }).isVisible();
+  await page.getByLabel("Description:").click();
+});
+
+test("creating a new freehand polygon opens a form", async () => {
+  await page.getByRole("button", { name: "New polygon (freehand)" }).click();
+  await clickMap(page, 500, 500);
+  await clickMap(page, 400, 500);
+  await clickMap(page, 400, 600);
+  await page.getByRole("button", { name: "Finish" }).click();
+
+  await page.getByRole("button", { name: "1) Untitled area" }).isVisible();
+  await page.getByLabel("Description:").click();
+});
+
+test("creating a new snapped polygon opens a form", async () => {
+  await page.getByRole("button", { name: "New polygon (snapped)" }).click();
+  await clickMap(page, 500, 500);
+  await clickMap(page, 400, 500);
+  await clickMap(page, 400, 600);
+  await page.getByRole("button", { name: "Finish" }).click();
+
+  await page.getByRole("button", { name: "1) Untitled area" }).isVisible();
+  await page.getByLabel("Description:").click();
+});
+
+test("creating a new route opens a form, and auto-fill sets its name", async () => {
+  await page.getByRole("button", { name: "New route" }).click();
+  await clickMap(page, 500, 500);
+  await clickMap(page, 400, 500);
+  await page.getByRole("button", { name: "Finish" }).click();
+
+  await page.getByRole("button", { name: "1) Untitled route" }).isVisible();
+  await page.getByLabel("Description:").click();
+
+  // This button only works after RouteInfo is loaded. And note because the
+  // button is located inside a label, getByRole doesn't seem to work.
+  await page.getByText("Auto-fill").click();
+  await page
+    .getByRole("button", {
+      name: "1) Route from ??? and Brighton Road to Emerald Quay and Harbour Way",
+    })
+    .isVisible();
+});
+
+test("editing geometry of a polygon works", async () => {
+  // Create a polygon
+  await page.getByRole("button", { name: "New polygon (snapped)" }).click();
+  await clickMap(page, 241, 579);
+  await clickMap(page, 235, 431);
+  await clickMap(page, 465, 459);
+  await page.getByRole("button", { name: "Finish" }).click();
+  await page.getByRole("button", { name: "1) Untitled area" }).isVisible();
+  await page.getByLabel("Description:").click();
+
+  // Click off the polygon to close the form
+  await clickMap(page, 284, 284);
+  // Make sure the form is closed
+  await expect(page.getByLabel("Description:")).not.toBeVisible();
+
+  // Click the polygon to open the form
+  await clickMap(page, 312, 501);
+  await page.getByLabel("Description:").click();
+
+  // Edit geometry, and click the polygon again
+  await page.getByRole("button", { name: "Edit geometry" }).click();
+  await clickMap(page, 312, 501);
+  // This button indicates the mode is working
+  await page.getByRole("button", { name: "Finish" }).click();
+});
+
+test("testing adding interventions, then deleting one, then adding another", async () => {
```

Comment:
These two came from the deleted files. Moved them here for better grouping of mode/drawing tests, and to share the worker